src/v/cluster/archival_metadata_stm.cc

```diff
@@ -880,7 +880,7 @@ archival_metadata_stm::get_segments_to_cleanup() const {
             // manifest in S3 so if we will delete it the data will be lost.
             if (m_name == s_name) {
                 vlog(
-                  _logger.warn,
+                  _logger.error,
```

Comment:
Could we place an alert on this log somehow? Would be good to know if/when it happens in production.